content/acra/security-controls/sql-prepared-statements/_index.md

```diff
@@ -0,0 +1,103 @@
+---
+title: SQL Prepared statements
+weight: 12
+---
+
+## Prepared statements from SQL
+
+{{< hint info >}}
+This feature is available in Acra only starting from 0.95.
+{{< /hint>}}
+
+Acra support alternative SQL interface to prepared statements
+for [`MySQL`](https://dev.mysql.com/doc/refman/8.0/en/sql-prepared-statements.html)
+and [`PostgreSQL`](https://www.postgresql.org/docs/current/sql-prepare.html).
+
+This interface is not as efficient as using the binary protocol through a prepared statement API, but requires no
+programming because it is available directly at the SQL level.
+
+### PostgreSQL
+
+AcraServer fully supports syntax of prepared statements for PostgreSQL.
+
+For Prepare query like:
+
+```
+PREPARE fooplan (int, text, bool, numeric) AS INSERT INTO foo VALUES($1, $2, $3, $4);
+```
+
+AcraServer will detect the prepared statement query and store it destructed for further processing.
+
+For Execute query like:
+
+```
+EXECUTE fooplan(1, 'Hunter Valley', 't', 200.00);
+```
+
+AcraServer will extract values based on the query stored in `Prepare` query and perform encryption/decryption.
+
+### MySQL
+
+MySQL has a different syntax for prepared statements.
+It uses `SET @a = 'value';` queries to provide data in DB and uses it as a reference in `EXECUTE` statements.
+
+To apply any security controls on data provided by `SET` queries. AcraServer expects it in the following format:
+
+```
+{table_from_encryptor_config}{delimiter}{column_from_encryptor_config}:
+```
+
+For example with query
+
+```
+SET @users__name = 'value'
+```
+
+and encryptor config:
+
+```
+schemas:
```

Comment:
let's change the order. at first show the encryptor config, after that show SET command. It is a bit easier to read and match in the mind, and logically correct. At first user should describe encryptor_config, then use this configuration in the own queries.